ros/angel_system_nodes/angel_system_nodes/activity_classification/activity_classifier_tcn.py

```diff
@@ -649,12 +660,29 @@ def rt_loop(self):
 
                         self._activity_publisher.publish(act_msg)
                     except NoActivityClassification:
+                        # collect the results if we are saving to coco file
+                        if self._results_collector:
+                            # Prepare output message
+                            activity_msg = ActivityDetection()
+                            # set the only needed items for collection
+                            activity_msg.source_stamp_end_frame = window.frames[-1][0]
+                            activity_msg.conf_vec = [0.0 for x in self._model.classes]
+                            self._collect_results(activity_msg)
```

Comment:
Why collect a zero-vector prediction for such a "failed" frame? I would think this is implied if there is an image in the output COCO file but no prediction. I am seeing that an image object was being added as part of the `collect` method, but I think I am realizing that is maybe more a historical artifact than a thing we should really be doing. Maybe the `ResultsCollector` class could be given an `add_image` method that we invoke before this try-catch to add the image metadata, and then only optionally collect the results annotation metadata on successful classification for that frame. Thoughts?

Reply:
I agree - I think that sounds like a better approach